CMakeLists.txt

```diff
@@ -63,6 +63,61 @@ set(WOLFSSL_INCLUDE_DIRS)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/")
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/functions.cmake)
 
+####################################################
+# Custom Functions
+####################################################
+
+# Function: SET_WOLFSSL_DEFINITIONS
+#   Parameter: SEARCH_VALUE  The string to search for. (e.g. "WOLFSSL_SHA3")
+#              OPERATION_KEYWORD     ["QUERY_ONLY" | "SET"]
+#   Returns:   RESULT
+#
+# Searches WOLFSSL_DEFINITIONS for SEARCH_VALUE
+#   Returns RESULT = 1 (true) if the search value is found
+#
+# Also sets a parent (global in this file) variable by the same name to 1.
+#
+function( SET_WOLFSSL_DEFINITIONS SEARCH_VALUE OPERATION_KEYWORD RESULT )
```

Comment:
Not sure what the advantages of adding this function are. It's only used to `SET` and not `QUERY` in the code. I can see how `QUERY` could be useful but for the `SET`, why not stick to the established pattern for adding macros to the build? Either way, it should probably be split into two functions and moved to `cmake/functions.cmake`

Reply:
> Not sure what the advantages of adding this function are.

I found there was a lot of duplicate code, so the function makes the code less verbose as well as easier to read and maintain.

> it should probably be split into two functions and moved

Agreed. I re-ran my [o-test.sh](https://github.com/gojimmypi/wolfssl/blob/ED25519_SHA2_fix/o-test.sh) script. (not included in this PR). The [output](https://github.com/gojimmypi/wolfssl/commits/ED25519_SHA2_fix/o-test.txt) was nearly identical.